docs/pre-commit-hooks.md

```diff
@@ -0,0 +1,110 @@
+# Pre-commit Hooks
+
+Git hooks are scripts that run automatically every time a particular event occurs in a Git repository. They let you customize Git’s internal behavior and trigger customizable actions at key points in the development life cycle. Git hooks are a useful way to run scripts when an action such as a commit, merge, or push occurs.
+
+To prevent the developer from committing bad code and running CI builds that would break the environment, we decided to use the pre-commit library to manage hooks and run some basic checks on the code. All the information about the pre-commit library can be found in the official documentation.
+
+The goal of this guide is to provide a quick starting point to the developer and explain how we are using it on the single pipeline repository and how we can configure it for other projects
+
+pre-commit runs locally while performing a commit, but with a small effort, it is possible to run it in CI too, during the CI build.
+
+Below are the few Hooks and their description,
+
+## Helm-deps (Helm Dependency Update)
+
+Manage the dependencies of a chart.
+
+```# Chart.yaml
+dependencies:
+- name: nginx
+  version: "1.2.3"
+  repository: "https://example.com/charts"
+- name: memcached
+  version: "3.2.1"
+  repository: "https://another.example.com/charts"
+```
+
+The 'name' should be the name of a chart, where that name must match the name in that chart's 'Chart.yaml' file.
+
+The 'version' field should contain a semantic version or version range.
+
+The 'repository' URL should point to a Chart Repository. Helm expects that by appending '/index.yaml' to the URL, it should be able to retrieve the chart repository's index. Note: 'repository' can be an alias. The alias must start with 'alias:' or '@'.
```

Comment:
```suggestion
Validate the dependencies of a chart.

Add to your `.pre-commit-config.yaml`:


  - repo: https://github.com/Alfresco/alfresco-build-tools
    rev: v1.22.0
    hooks:
      - id: helm-deps

```

Reply:
provides a similar structure for all the hooks defined in https://github.com/Alfresco/alfresco-build-tools/blob/master/.pre-commit-hooks.yaml